cmd/eval.go

```diff
@@ -88,3 +36,52 @@ func validateFile(file []byte) error {
 	var holder interface{}
 	return yaml.Unmarshal(file, &holder)
 }
+
+var (
+	evalCmd = &cobra.Command{
+		Use:   "eval [file]",
+		Short: "Evaluate the combinations for a file at the given path",
+		Long: `
+Evaluate the combinations for a file at the given path. The file provided must be valid YAML.
+
+Note: the combo binary requires the --replacement flag to be explicitly set.
+
+The replacements flag allows users to specify a series of key value pairs in the form of KEY=VALUES.
+
+Example: combo eval -r REPLACE_ME=1,2,3 path/to/file
+	`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			replacements, _ := cmd.Flags().GetStringToString("replacements")
+
+			file, err := os.ReadFile(args[0])
+			if err != nil {
+				return err
+			}
+
+			if err := validateFile(file); err != nil {
+				return fmt.Errorf("failed to validate file specified: %w", err)
+			}
+
+			combinations := combination.NewStream(
+				combination.WithArgs(formatReplacements(replacements)),
+				combination.WithSolveAhead(),
+			)
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			generatedFile, err := generator.Evaluate(ctx, string(file), combinations)
+			if err != nil {
+				return err
+			}
+
+			if err := validateFile([]byte(generatedFile)); err != nil {
+				return fmt.Errorf("failed to validate file generated: %w", err)
+			}
+
+			fmt.Println(generatedFile)
```

Comment:
I am going to assume that the actual logic of the controller in which the generated resources are applied on cluster will appear in a follow up PR. As such, I will not be reviewing the controller logic at this time.